performance-test/src/puppeteer-test.ts

```diff
@@ -47,33 +64,123 @@ export const testScrollingPerformance = async function () {
   await page.goto(EDITOR_URL)
+  return {
+    browser: browser,
+    page: page,
+  }
+}
+
+export const testPerformance = async function () {
+  const scrollResult = await testScrollingPerformance()
+  const resizeResult = await testResizePerformance()
+  const selectionResult = await testSelectionPerformance()
+
+  const scrollImage = await uploadImage(scrollResult)
+  const resizeImage = await uploadImage(resizeResult)
+  const selectionImage = await uploadImage(selectionResult, true)
+
+  console.info(
+    `::set-output name=perf-result:: ![ScrollChart](${scrollImage}) - SCROLL TEST Average frame length: ${scrollResult.frameData.frameAvg.toFixed(
+      1,
+    )} – Q1: ${scrollResult.frameData.percentile25} – Q2: ${
+      scrollResult.frameData.percentile50
+    } – Q3: ${scrollResult.frameData.percentile75} – Median: ${
+      scrollResult.frameData.percentile50
+    } ![ResizeChart](${resizeImage}) - RESIZE TEST Average frame length: ${resizeResult.frameData.frameAvg.toFixed(
+      1,
+    )} – Q1: ${resizeResult.frameData.percentile25} – Q2: ${
+      resizeResult.frameData.percentile50
+    } – Q3: ${resizeResult.frameData.percentile75} – Median: ${
+      resizeResult.frameData.percentile50
+    } ![SelectionChart](${selectionImage}) - SELECTION TEST Average frame length: ${selectionResult.frameData.frameAvg.toFixed(
+      1,
+    )} – Q1: ${selectionResult.frameData.percentile25} – Q2: ${
+      selectionResult.frameData.percentile50
+    } – Q3: ${selectionResult.frameData.percentile75} – Median: ${
+      selectionResult.frameData.percentile50
+    }`,
+  )
+}
+
+export const testScrollingPerformance = async function (): Promise<FrameResult> {
+  const { page, browser } = await setupBrowser()
   await page.waitForXPath("//a[contains(., 'P S')]") // the button with the text 'P S' is the "secret" trigger to start the scrolling performance test
   // we run it twice without measurements to warm up the environment
   const [button] = await page.$x("//a[contains(., 'P S')]")
   await button!.click()
-  await consoleDoneMessage(page)
+  await consoleDoneMessage(page, 'SCROLL_TEST_FINISHED')
   const [button2] = await page.$x("//a[contains(., 'P S')]")
   await button2!.click()
-  await consoleDoneMessage(page)
+  await consoleDoneMessage(page, 'SCROLL_TEST_FINISHED')
   // and then we run the test for a third time, this time running tracing
   await page.tracing.start({ path: 'trace.json' })
   const [button3] = await page.$x("//a[contains(., 'P S')]")
   await button3!.click()
-  await consoleDoneMessage(page)
+  await consoleDoneMessage(page, 'SCROLL_TEST_FINISHED')
   await page.tracing.stop()
   await browser.close()
   let traceData = fs.readFileSync('trace.json').toString()
   const traceJson = JSON.parse(traceData)
 
+  return getFrameData(traceJson, 'scroll_step_')
+}
+
+export const testResizePerformance = async function (): Promise<FrameResult> {
+  const { page, browser } = await setupBrowser()
```

Comment:
All of these tests need to be wrapped in a `try` `finally` block to ensure that `browser.close()` is still called if they throw an error. I know Omar made that change locally last week whilst I was pairing up with him, but it looks like it wasn't merged.

Reply:
fixed